src/darwin/Framework/CHIP/MTRDeviceControllerFactory.mm

```diff
@@ -559,7 +629,7 @@ - (void)stopControllerFactory
  * return nil if pre-startup fabric table checks fail, and set fabricError to
  * the right error value in that situation.
  */
-- (MTRDeviceController * _Nullable)_startDeviceController:(MTRDeviceControllerStartupParams *)startupParams
+- (MTRDeviceController * _Nullable)_startDeviceController:(id)startupParams
```

Comment:
Seems like startup parameters should not be an id here, so we can at least say what class derivatives are acceptable?

Reply:
I guess this is needed to fork the old from the new though for now....

Reply:
Right, exactly.  This is all internal bits and we in practice only have two possible things that might be coming through.